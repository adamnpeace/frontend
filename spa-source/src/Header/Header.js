import React from "react";
import { Link } from "react-router-dom";
import logo from "../img/logo.svg";

const Header = () => (
  <nav className="navbar navbar-expand-md navbar-light navbar-static-top bg-primary ">
    <a href="/" className="navbar-brand">
      <img src={logo} alt="logo" height="24px" />
    </a>
    <div className="navbar-collapse collapse" id="navbarColor01">
      <ul className="navbar-nav mr-auto">
        <li className="nav-item">
          <a href="/" className="nav-link">
            Explore
          </a>
        </li>
        <li className="nav-item">
          <Link to="#" className="nav-link">
            Labs
          </Link>
        </li>
      </ul>
    </div>
  </nav>
);

export default Header;
