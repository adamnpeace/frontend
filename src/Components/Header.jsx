import React from 'react';
import { Box } from 'grommet';

const Header = props => (
  <Box
    align="center"
    background="brand"
    direction="row"
    elevation="medium"
    height="60px"
    justify="between"
    pad={{ left: 'medium', right: 'small', vertical: 'small' }}
    style={{ zIndex: '1' }}
    tag="header"
    {...props}
  />
);

export default Header;
