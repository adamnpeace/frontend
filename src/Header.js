import React from "react";
import { Box } from "grommet";

const Header = props => (
  <Box
    align="center"
    background="brand"
    direction="row"
    elevation="medium"
    height="7vh"
    justify="between"
    pad={{ left: "medium", right: "small", vertical: "small" }}
    style={{ zIndex: "1" }}
    tag="header"
    {...props}
  />
);

export default Header;
