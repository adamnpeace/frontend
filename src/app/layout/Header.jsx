import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { NavLink, withRouter } from 'react-router-dom';
import { notify } from 'react-notify-toast';
import Firebase from '../../auth/firebase';
import { useGlobalState } from '../../state';

import { Text } from '../../components/logo';
import logo from '../../img/logos/text_black.svg';

function Header(props) {
  const [isActive, setIsActive] = useState(false);
  const [isAuthenticated, setIsAuthenticated] = useGlobalState('auth');
  const [name, setName] = useGlobalState('name');

  function handleLogout() {
    Firebase.logout();
    setIsAuthenticated(false);
    notify.show('You have been logged out successfully!', 'warning');
    props.history.replace('/');
  }

  return (
    <nav
      className="navbar"
      aria-label="main navigation"
      style={{
        borderBottom: 'solid 1px #dddddd'
      }}
    >
      <div className="navbar-brand">
        <NavLink className="navbar-item" to="/" activeClassName="is-active">
          <img src={logo} alt="home" />
        </NavLink>
        <button className="button navbar-burger" onClick={() => setIsActive(!isActive)}>
          <span />
          <span />
          <span />
        </button>
      </div>
      <div className={isActive ? 'navbar-menu is-active' : 'navbar-menu'}>
        <div className="navbar-start">
          <NavLink
            className="navbar-item"
            to={isAuthenticated ? '/topics' : '/login'}
            activeClassName="is-active"
          >
            <span className="icon" style={{ marginRight: 5 }}>
              <i className="fas fa-lg fa-graduation-cap" />
            </span>
            Learn
          </NavLink>
        </div>
        {isAuthenticated ? (
          <div class="navbar-end">
            <div class="navbar-item has-dropdown is-hoverable">
              <div class="navbar-link">
                <span className="icon" style={{ marginRight: 5 }}>
                  <i className="fas fa-lg fa-user-circle" />
                </span>
                My Account
              </div>

              <div class="navbar-dropdown">
                <div class="navbar-item">Hi, {name}</div>
                <hr class="navbar-divider" />
                <a className="navbar-item" onClick={handleLogout}>
                  <span className="icon" style={{ marginRight: 5 }}>
                    <i className="fas fa-lg fa-sign-in-alt" />
                  </span>
                  Log out
                </a>
              </div>
            </div>
          </div>
        ) : (
          <div class="navbar-end">
            <div class="navbar-item">
              <div class="buttons">
                <NavLink to="/signup" className="button is-primary">
                  <strong>Sign Up</strong>
                </NavLink>
                <NavLink to="/login" className="button is-light">
                  Log in
                </NavLink>
              </div>
            </div>
          </div>
        )}
      </div>
    </nav>
  );
}

export default withRouter(Header);
