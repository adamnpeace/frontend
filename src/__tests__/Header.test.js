import React from "react";
import ReactDOM from "react-dom";
import Header from '../Header';
import renderer from "react-test-renderer";

describe("Editor Component Test", () => {
  it("Header renders without crashing", () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it("Header renders correctly according to first snapshot", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
