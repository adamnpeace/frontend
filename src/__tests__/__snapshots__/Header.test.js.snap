// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Component Test Header renders correctly according to first snapshot 1`] = `
<header
  className="StyledBox-sc-13pk1d4-0 bRQYIV"
  style={
    Object {
      "zIndex": "1",
    }
  }
/>
`;
